src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1652,6 +1656,9 @@ private void ImportThrow()
 
         private LLVMValueRef GetInstanceFieldAddress(StackEntry objectEntry, FieldDesc field)
         {
+            if (!field.IsTypicalFieldDefinition)
```

Comment:
No, what I meant is that the call to `GetTypicalFieldDefinition` didn't look right.

Codegen should never operate on uninstantiated things if generics are involved. Everything should be fully instantiated. We can't compute field offsets, can't determine field sizes, etc. Uninstantiated things are useless.

Typical field definition is a field definition on the type definition. I.e. if Foo is nongeneric, Foo.Field is a typical field definition. If Foo is generic, Foo.Field is a typical field definition, Foo<Something>.Field is not a typical field definition.

The new throw doesn't look right.

Reply:
I switched it to `if (field.FieldType.IsSignatureVariable)` as that seems to be a more direct representation of my problem, but I die later when a Internal.TypeSystem.MethodForInstantiatedType fails for obvious reasons to get a mangled name. My problem here is I don't understand why all of these open generics are suddenly flying around in codegen. How are my UnboxingStub changes causing this?

Reply:
The problem is definately inside GetUnboxingStub
```
var owningTypeDefinition = (MetadataType)owningType.GetTypeDefinition();
```
```
if (owningType != owningTypeDefinition)
{
    InstantiatedType boxedType = boxedTypeDefinition.MakeInstantiatedType(owningType.Instantiation);
    MethodDesc thunk = GetMethodForInstantiatedType(thunkDefinition, boxedType);
    Debug.Assert(!thunk.HasInstantiation);
    return thunk;
}
```

this is where the Internal.TypeSystem.MethodForInstantiatedType is coming from. Why am i not just using the owningType directly, I think I may have just misunderstood your initial comment about this area.

Reply:
You missed the `InstantiateAsOpen()` call in the method body generator.

The generated IL for methods on generic types is:

```
// [S.P.CompilerGenerated]Boxed_Enumerator<string>.get_Current_Unbox()
.method instance string get_Current_Unbox() cil managed
{
  // Code size: 12
  .maxstack 12

  IL_0000:  ldarg.0
  IL_0001:  ldflda      valuetype [System.Private.CoreLib]System.Collections.Generic.List`1/Enumerator<string> class Boxed_Enumerator<string>::BoxedValue
  IL_0006:  call        instance !0 valuetype [System.Private.CoreLib]System.Collections.Generic.List`1/Enumerator::get_Current()
  IL_000B:  ret
}
```

But it should be:

```
// [S.P.CompilerGenerated]Boxed_Enumerator<string>.get_Current_Unbox()
.method instance string get_Current_Unbox() cil managed
{
  // Code size: 12
  .maxstack 12

  IL_0000:  ldarg.0
  IL_0001:  ldflda      valuetype [System.Private.CoreLib]System.Collections.Generic.List`1/Enumerator<string> class Boxed_Enumerator<string>::BoxedValue
  IL_0006:  call        instance string valuetype [System.Private.CoreLib]System.Collections.Generic.List`1/Enumerator<string>::get_Current()
  IL_000B:  ret
}
```